verification/spec/connection-handshake/ch.tla

```diff
@@ -0,0 +1,188 @@
+
+Null == "none"
+
+ConnectionIDs == {"connAtoB", "connBtoA"}
+
+
+ClientIDs == { "clientA", "clientB" }
+
+
+(***************************** ConnectionEnds *****************************
+    Ref.: https://github.com/informalsystems/ibc-rs/pull/55.
+    
+    A set of connection end records. 
+    A connection end record contains the following fields:
+    
+    - state -- a string 
+      Stores the current state of this connection end. It has one of the 
+      following values: "UNINIT", "INIT", "TRYOPEN", "OPEN".
+    
+    - connectionID -- a connection identifier
+      Stores the connection identifier of this connection end.
+    
+    - counterpartyConnectionID -- a connection identifier
+      Stores the connection identifier of the counterparty connection end.
+    
+    - clientID -- a client identifier
+      Stores the client identifier associated with this connection end. 
+      
+    - counterpartyClientID -- a client identifier
+      Stores the counterparty client identifier associated with this connection end.
+ ***************************************************************************)
+ConnectionEnd ==
+    [
+        state : ConnectionStates,
+        connectionID : ConnectionIDs \union {Null},
+        clientID : ClientIDs \union {Null},
+        counterpartyConnectionID : ConnectionIDs \union {Null},
+        counterpartyClientID : ClientIDs \union {Null}
+    ]      
+
+(********************************** Helper functions
+ ***************************************************************************)
+
+\* Constructs a ConnectionEnd object for the given party in state UNINIT.
+getConnectioEnd(targetParty) ==
+    IF targetParty = "A"
+    THEN 
+        [   state |-> "UNINIT",
+            connectionID |-> "connAtoB",
+            clientID |-> "clientA",
+            counterpartyConnectionID |-> "connBtoA",
+            counterpartyClientID |-> "clientB" ]
+    ELSE IF targetParty = "B"
+         THEN
+            [   state |-> "UNINIT",
+                connectionID |-> "connBtoA",
+                clientID |-> "clientB",
+                counterpartyConnectionID |-> "connAtoB",
+                counterpartyClientID |-> "clientA" ]
+         ELSE Null    
+
+
+\* Constructs a 'Init' Datagram.
+\* Fields are hardcoded for the moment
+getInitDatagram(targetParty) ==
+    [ type |-> "ConnOpenInit", 
+      connectionEnd |-> getConnectioEnd(targetParty) ]
+
+
+isPartyInState(p, state) ==
+    parties[p].connectionEnd.state = state
+
+
+\* Would be ideal to import the following function.
+\* Cf. ibc-rs#55 (file: ConnectionHandlers.tla)
+getConnectionID(p) ==
+    IF p = "A"
+    THEN "connAtoB"
+    ELSE IF p = "B"
+         THEN "connBtoA"
+         ELSE Null  
+
+
+(********************************** Environment functions
+ ***************************************************************************)
+
+    
+\* The environment generates the ConnOpenInit datagram, sending it to 'party'
+GenerateConnOpenInit(targetParty) ==
```

Comment:
I think that environment should be super simple in this spec; it does not need any variable or helper functions. It just set inMsg variable to any value from the set of InMsgs that we need to define (InitConn, TryConnOpen, etc).  For example `inMsg \in InMsgs`.